closure/goog/net/jsloader.js

```diff
@@ -94,36 +94,26 @@ goog.net.jsloader.scriptsToLoad_ = [];
  * @param {Array<string>} uris The URIs to load.
  * @param {goog.net.jsloader.Options=} opt_options Optional parameters. See
  *     goog.net.jsloader.options documentation for details.
+ * @return {!goog.async.Deferred} The deferred result, that may be used to add
+ *     callbacks
  */
 goog.net.jsloader.loadMany = function(uris, opt_options) {
   // Loading the scripts in serial introduces asynchronosity into the flow.
-  // Therefore, there are race conditions where client A can kick off the load
-  // sequence for client B, even though client A's scripts haven't all been
-  // loaded yet.
-  //
-  // To work around this issue, all module loads share a queue.
+  // The client receives a deferred to manage parallel loads.
   if (!uris.length) {
-    return;
-  }
-
-  var isAnotherModuleLoading = goog.net.jsloader.scriptsToLoad_.length;
-  goog.array.extend(goog.net.jsloader.scriptsToLoad_, uris);
-  if (isAnotherModuleLoading) {
-    // jsloader is still loading some other scripts.
-    // In order to prevent the race condition noted above, we just add
-    // these URIs to the end of the scripts' queue and return.
-    return;
```

Comment:
>But since we always return deferreds now, that use-case could easily be handled by the caller.

I don't know if I agree with this. Yes, it's possible for the caller to manage it, but it would require passing around a deferred to anything that wants to chain loads. The way it is now, your only dependency is on `jsloader`, and invoking `loadMany` means you are always guaranteed to get the latest deferred regardless of where/how you call it. IMO, this seems nicer.

Re: backwards compatibility, this is the cost of maintaining a widely used library. You have to weigh the cost and potential payoff of a change. If we make this change and I break 200 tests, is it really worth me spending 3 days fixing them for this change? Especially when it's just as easy to just add an option to enable the functionality you want without breaking anyone.